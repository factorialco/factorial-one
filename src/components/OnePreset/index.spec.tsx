import { render, screen, userEvent } from "@testing-library/react-native";
import React from "react";
import { OnePreset } from ".";

describe("OnePreset", () => {
  const defaultProps = {
    label: "label",
  };

  it("Snapshot", () => {
    const { toJSON } = render(<OnePreset {...defaultProps} />);
    expect(toJSON()).toMatchSnapshot();
  });

  it("Snapshot with number", () => {
    const { toJSON } = render(<OnePreset {...defaultProps} number={42} />);
    expect(toJSON()).toMatchSnapshot();
  });

  it("renders correctly with selected props", () => {
    const { toJSON } = render(<OnePreset {...defaultProps} selected={true} />);

    expect(toJSON()).toMatchSnapshot();
  });

  it("renders correctly with required props", () => {
    render(<OnePreset {...defaultProps} />);

    const number = screen.getByText("label");

    expect(number).toBeDefined();
  });

  it("renders correctly with onClick event", async () => {
    const testFn = jest.fn();
    const user = userEvent.setup();

    render(<OnePreset {...defaultProps} onClick={testFn} />);

    const number = screen.getByText("label");
    await user.press(number);

    expect(testFn).toHaveBeenCalledTimes(1);
    expect(number).toBeDefined();
  });
});
