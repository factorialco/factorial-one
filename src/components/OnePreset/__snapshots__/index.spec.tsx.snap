// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OnePreset Snapshot 1`] = `
<View
  className="flex items-start"
>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    className="flex grow-0 flex-row items-center gap-2 rounded border border-f1-border px-2.5 py-1.5 font-medium text-f1-foreground"
    collapsable={false}
    focusable={true}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
  >
    <Text
      className="whitespace-nowrap text-f1-foreground"
    >
      label
    </Text>
  </View>
</View>
`;

exports[`OnePreset Snapshot with number 1`] = `
<View
  className="flex items-start"
>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    className="flex grow-0 flex-row items-center gap-2 rounded border border-f1-border px-2.5 py-1.5 font-medium text-f1-foreground pr-1.5"
    collapsable={false}
    focusable={true}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
  >
    <Text
      className="whitespace-nowrap text-f1-foreground"
    >
      label
    </Text>
    <View
      className="flex items-start"
    >
      <View
        className="flex items-center justify-center rounded-xs grow-0 px-0.5 min-w-5 h-5 bg-f1-background-secondary border border-f1-border"
      >
        <Text
          className="text-center text-sm font-medium tabular-nums whitespace-nowrap text-f1-foreground"
        >
          42
        </Text>
      </View>
    </View>
  </View>
</View>
`;

exports[`OnePreset renders correctly with selected props 1`] = `
<View
  className="flex items-start"
>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    className="flex grow-0 flex-row items-center gap-2 rounded border px-2.5 py-1.5 font-medium border-f1-border-selected bg-f1-background-selected-secondary text-f1-foreground-selected"
    collapsable={false}
    focusable={true}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
  >
    <Text
      className="whitespace-nowrap text-f1-foreground"
    >
      label
    </Text>
  </View>
</View>
`;
